src/gaxios.ts

```diff
@@ -318,25 +318,30 @@ export class Gaxios {
     }
     opts.method = opts.method || 'GET';
 
-    const proxy = getProxy(opts.url);
-    if (proxy) {
+    const shouldUseProxy = this.#shouldUseProxyForURLIfAvailable(
+      opts.url,
+      opts.noProxy
+    );
+
+    const proxy =
+      opts.proxy ||
+      process?.env?.HTTPS_PROXY ||
+      process?.env?.https_proxy ||
+      process?.env?.HTTP_PROXY ||
+      process?.env?.http_proxy;
+
+    if (shouldUseProxy && proxy) {
+      const HttpsProxyAgent = await Gaxios.#getProxyAgent();
+
       if (this.agentCache.has(proxy)) {
         opts.agent = this.agentCache.get(proxy);
       } else {
-        // Proxy is being used in conjunction with mTLS.
-        if (opts.cert && opts.key) {
-          const parsedURL = new URL(proxy);
-          opts.agent = new HttpsProxyAgent({
-            port: parsedURL.port,
-            host: parsedURL.host,
-            protocol: parsedURL.protocol,
-            cert: opts.cert,
-            key: opts.key,
-          });
-        } else {
-          opts.agent = new HttpsProxyAgent(proxy);
-        }
-        this.agentCache.set(proxy, opts.agent!);
+        opts.agent = new HttpsProxyAgent(proxy, {
+          cert: opts.cert,
+          key: opts.key,
+        });
```

Comment:
Looking at the following link I do not believe mTLS was working for proxied requests previously:
- https://github.com/TooTallNate/proxy-agents/blob/010fdc304a18931d2a1da7767e71303e7ce05abd/packages/https-proxy-agent/src/index.ts#L51-L75

This was due to `HttpsProxyAgent` being `any` at the time

Reply:
Woah, if I'm reading this right how did it work at all? Looks like the first parameter should have been `Uri | URL`, not an object. Nice catch.

Reply:
Yep! Never worked at all 😔